less_users/tests/test_user_profile.py

```diff
@@ -1,9 +1,33 @@
-from less_users.models import Profile
+from less_users.models import Profile, UserChallenge, Log
 
 
-def test_profile_user_create(user):
+def test_create_user_profile(user):
     profile = user.profile
 
-    assert profile in Profile.objects.all()
+    assert Profile.objects.filter(id=profile.id).exists()
     assert user.profile.points == 0
 
+
+def test_change_user_profile(user):
+    profile = user.profile
+    profile.points = 10
+
+    assert user.profile.points == 10
+
+
+def test_change_user_profile_by_user_update_form(user):
+    pass
+
+
+def test_change_user_profile_points_by_clicking_done(user, challenge_2_week):
```

Comment:
niezbyt wiem co chcesz tutaj przetestować

Reply:
jak wywołam funkcję w views.py o nazwie event_view_done, to ona  po drodze tworzy Loga, a ostatecznie nalicza punkty dla profilu Usera. Pod warunkiem spełnienia założeń: jest user, jest challenge, jest user_challenge, które jest aktywne, i nie było wcześniejszego loga w mniejszej odległości dni niż określone 'frequency'

Reply:
No dobra, tylko:
- nie robisz tutaj requestu do widoku;
- jak dotyczy widoku to nie powinno być w pliku `test_user_profile.py`;
- Test powinien opowiadać historię - przy takich założeniach wejściowych (jest user, jest user challenge, challenge daje x punktów), po wywołaniu jakiejś funkcji warunki wyjściowe bedą takie a takie (np user ma o x punktów więcej). Tutaj masz bardziej logikę: jeżeli a i b są spełnione to zrób c a potem sprawdzasz czy c jest aby na pewno c.

Reply:
ok, czyli zawsze "arrange-act-assert" ma sens ;)

Reply:
widzę, że nie mam tu sprawdzonego mechanizmu z views.py, bo tylko nadpisuję dane do tymczasowej bazy danych. Ale nie wiem jak powołać się na funkcję z views.py w testach...

Reply:
> widzę, że nie mam tu sprawdzonego mechanizmu z views.py, bo tylko nadpisuję dane do tymczasowej bazy danych. Ale nie wiem jak powołać się na funkcję z views.py w testach...


z czym konkretnie masz problem? jakich sposobów próbowałaś?

Reply:
To też już można wywalić

Reply:
Tak, już mi wytłumaczyłaś jak robić testy widoków :)

Reply:
Czy mogłabyś nie resolvować konwersacji?
https://www.danclarke.com/resolving-pr-comments